packages/@ama-sdk/schematics/schematics/index.it.spec.ts

```diff
@@ -0,0 +1,97 @@
+
+/**
+ * Set up a local npm registry inside a docker image before the tests.
+ * Publish all the packages of the Otter monorepo on it.
+ * Can be accessed during the tests with url http://localhost:4873
+ */
+function setupLocalRegistry() {
+  let shouldHandleVerdaccio = false;
+
+  beforeAll(async () => {
+    try {
+      await getPidFromPort(4873);
+    } catch (ex) {
+      shouldHandleVerdaccio = true;
+      execSync('yarn verdaccio:start', {cwd: currentFolder, stdio: 'inherit'});
+      execSync('yarn verdaccio:publish', {cwd: currentFolder, stdio: 'inherit'});
+    }
+  });
+
+  afterAll(() => {
+    if (shouldHandleVerdaccio) {
+      execSync('yarn verdaccio:stop', {cwd: currentFolder, stdio: 'inherit'});
+    }
+  });
+}
+
+function setupYarn(yarnVersion: string) {
+  execSync('yarn config set enableStrictSsl false', execAppOptions);
+  execSync(`yarn set version ${yarnVersion}`, execAppOptions);
+  execSync(`yarn config set npmScopes.ama-sdk.npmRegistryServer ${registry}`, execAppOptions);
+  execSync(`yarn config set npmScopes.o3r.npmRegistryServer ${registry}`, execAppOptions);
+  execSync('yarn config set unsafeHttpWhitelist localhost', execAppOptions);
+  execSync('yarn config set nodeLinker pnp', execAppOptions);
+  execSync(`yarn config set cacheFolder ${path.posix.join(currentFolder, '.cache', 'test-sdk')}`, execAppOptions);
+  execSync('yarn config set enableImmutableInstalls false', execAppOptions);
+}
+
+/**
+ * Setup a new sdk using schematics CLI
+ */
+function setupNewSdk() {
+  beforeAll(() => {
+    const packageJson = JSON.parse(readFileSync(packageJsonPath).toString()) as PackageJson;
+    const angularVersion = minVersion(packageJson.devDependencies['@angular-devkit/schematics-cli']).version;
+
+    // Create app with ng new
+    const relativePath = path.posix.relative(parentFolderPath, sdkFolderPath);
+    execSync(`npx rimraf ${relativePath}`, {cwd: parentFolderPath, stdio: 'inherit'});
+    execSync(`mkdir -p ${relativePath}`, {cwd: parentFolderPath, stdio: 'inherit'});
+    writeFileSync(path.posix.join(sdkFolderPath, 'package.json'), '{"name": "@test/sdk"}');
+
+    // Set config to target local registry
+    const o3rPackageJson: PackageJson & { packageManager?: string } = JSON.parse(fs.readFileSync(path.posix.join(currentFolder, 'package.json')).toString());
+    const yarnVersion = o3rPackageJson?.packageManager?.split('@')?.[1] || '3.5.0';
+    setupYarn(yarnVersion);
+    // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+    execSync(`yarn add -D @angular-devkit/schematics-cli@${angularVersion} @ama-sdk/schematics`, execAppOptions);
+    cpSync(path.posix.join(__dirname, '..', 'testing', 'MOCK_swagger.yaml'), path.posix.join(sdkFolderPath, 'swagger-spec.yml'));
+    execSync('yarn schematics @ama-sdk/schematics:typescript-sdk --name test --package sdk --swagger-spec-path ./swagger-spec.yml', execAppOptions);
+    setupYarn(yarnVersion);
```

Comment:
do you need to run it again ?

Reply:
Yes because the schematics setup the repository but is not installing all the deps from the new package.json that override the one created in the test